### ISSUE TITLE
Cumulocity IoT firmware update command should include a locally accessible tedgeUrl

### ISSUE THREAD

**Issue by reubenmiller** [thin-edge/thin-edge.io#2793]
**Is your feature improvement request related to a problem? Please describe.**

thin-edge.io **firmware_update** operations do not translate Cumulocity IoT URLs provided in the `remoteUrl` property to a locally accessible c8y proxy URL (e.g. `http://rpi5-d83add9f145a.local:8001/c8y/inventory/binaries/46252`). This makes it more difficult for clients to download the resources as it requires the clients to either request a Cumulocity IoT token, or to modify the URL to a local c8y proxy (which can be more difficult in some other languages like C).

Below shows an example of a child `firmware_update` command received by a microcontroller client (notice the `remoteUrl` URL contains the full Cumulocity IoT URL).

```sh
te/device/esp32-e465b86f97cc///cmd/firmware_update/c8y-mapper-52552 {
    "status": "init",
    "remoteUrl": "https://example.c8y.io/inventory/binaries/51541",
    "name": "esp32-tedge",
    "version": "1.1.1"
}
```

Where as the url provided in a `software_update` command does translate a public Cumulocity IoT URL to a local c8y proxy URL.

```sh
te/device/esp32-e465b86f97cc///cmd/software_update/c8y-mapper-52554 {
    "status": "init",
    "updateList": [
        {
            "type": "esp32",
            "modules": [
                {
                    "name": "esp-client",
                    "version": "1.2.3",
                    "url": "http://rpi5-d83add9f145a.local:8001/c8y/inventory/binaries/46252",
                    "action": "install"
                }
            ]
        }
    ]
}
```

**Describe the solution you'd like**

The `tedge-mapper-c8y` service should add an additional `tedgeUrl` property to the `firmware_update` command payload where the `remoteUrl` is translated to a local tedge URL if the `remoteUrl` points to the Cumulocity IoT URL (the same Cumulocity IoT instance that the mapper is connected to).

**Describe alternatives you've considered**

**Additional context**

**Comment by rina23q** [comment]
Resolved by #2795

**Comment by gligorisaev** [comment]
QA has thoroughly checked the feature and here are the results:

- [ ] Test for ticket exists in the test suite.
- [x] QA has tested the function and it's functioning according description.
